docs/DeveloperGuide.adoc

```diff
@@ -30,58 +30,68 @@ image::ArchitectureDiagram.png[]
 
 * At app launch: Initializes the components in the correct sequence, and connects them up with each other.
 * At shut down: Shuts down the components and invokes cleanup method where necessary.
 
-<<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.
+* <<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.
 The following class plays an important role at the architecture level:
 
-* `LogsCenter` : Used by many classes to write log messages to the App's log file.
+* `LogsCenter` is used by many classes to write log messages to the App's log file.
 
 The rest of the App consists of four components.
 
 * <<Design-Ui,*`UI`*>>: The UI of the App.
-* <<Design-Logic,*`Logic`*>>: The command executor.
-* <<Design-Model,*`Model`*>>: Holds the data of the App in-memory.
+* <<Design-Logic,*`Logic`*>>: The command executor depending on the view the user is on.
+* <<Design-Model,*`Model`*>>: Holds the data of the App in-memory dependong in the view the user is on.
 * <<Design-Storage,*`Storage`*>>: Reads data from, and writes data to, the hard disk.
 
 Each of the four components
 
 * Defines its _API_ in an `interface` with the same name as the Component.
-* Exposes its functionality using a `{Component Name}Manager` class.
+* Exposes its functionality using a `{Component Name} Manager` class.
 
 For example, the `Logic` component (see the class diagram given below) defines it's API in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class.
 
-.Class Diagram of the Logic Component
-image::LogicClassDiagram.png[]
+There are 4 views in this App: `HOME`, `TRIP`, `ITINERARY`, `JOURNAL`.
+
+For the `Logic` and `Model` components, each view has their own `{Component Name} Manager` classes.
+
+e.g. `JournalLogicManager` etc. to facilitate different operations and commands in different views in the App.
+
+
+.Class Diagram of the Journal Logic Component
+image::JournalLogicDiagram.png[]
 
 [discrete]
 ==== How the architecture components interact with each other
 
-The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `back` when in a `TRIP` view.
 
-.Component interactions for `delete 1` command
+.Component interactions for `back` command in the `TRIP` view.
 image::ArchitectureSequenceDiagram.png[]
 
-The sections below give more details of each component.
+
+The _Sequence Diagram_ below illustrates how the components iteract with each other for the scenario where the user issues the command `goto 1` when in a `HomePage` view.
+
+.Component interactions for `goto 1` command in the `HomePage` view.
```

Comment:
The "new" can be removed. Method calls should have (). From user to mainwindow, it should be a method call.